### DISCUSSION TITLE
How to implement a custom filter to hide products

### DISCUSSION THREAD

**Discussion by onepack** [Smile-SA/elasticsuite#3169] [Q&A]
Hi there,

I have developed a module that can filter products based on customer groups.
After login you would only see the customer group assigned products.

This works fine without the Smile Elastic Suite module directly extending elastic search 7.

My question is what Class would I need to use instead of "Magento\Elasticsearch7\Model\Client\Elasticsearch" in case I use the Smile Elasticsuite in my di.xml. I've been trying some classes but have been unlucky until now.

`<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:ObjectManager/etc/config.xsd">
    <type name="Magento\Elasticsearch7\Model\Client\Elasticsearch">
        <plugin name="opck_customcollection_elastic_product_index" type="Opck\FilterProducts\Plugin\ElasticLayerPlugin" />
    </type>
</config>`

**Comment by romainruaud** [comment]
Hi, this should not be a plugin on the client.

With Elasticsuite, this should be a "filter", as it's done for the stock : 

https://github.com/Smile-SA/elasticsuite/blob/2.11.x/src/module-elasticsuite-catalog/etc/elasticsuite_search_request.xml#L23

And the underlying implementation : 

https://github.com/Smile-SA/elasticsuite/blob/2.11.x/src/module-elasticsuite-catalog/Model/Product/Search/Request/Container/Filter/Stock.php

Regards

**Comment by onepack** [comment]
I will give it a try.
Just in case I was not specific enough. The client should NOT see the products at all that are not assigned to his group.
So the collection is not based on a filter option on the sidebar but rather based of the assignment of the products to a customer group that is allowed to see it (or not). So after logging in they only see the assigned collection.

This works now but I really want to try and make it work with Elastic Suite as well.
